rpc/client.go

```diff
@@ -109,6 +148,106 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+// getSignedUrl calls the Data Pipes API to retrieve a signed URL, where it can PUT the report json.
+// Data Pipes docs: https://github.com/10gen/data-pipes
+func getSignedURL(task string, execution int, data []byte, AWSAccessKey string, AWSSecretKey string, AWSToken string, dataPipesHost string, dataPipesRegion string) (string, error) {
+	service := "execute-api"
+	resultType := "cedar-report"
+	if AWSAccessKey == "" || AWSSecretKey == "" || dataPipesHost == "" || dataPipesRegion == "" {
+		return "", errors.New("Getting signed URL failed. AWS access key, AWS secret key, data pipes host and data pipes region required.")
+	}
+	client := &http.Client{}
```

Comment:
Use the client pool here:

```
client := utility.GetHTTPClient()
defer utility.PutHTTPClient()
```

Reply:
This ended up making it tricky to mock, so I passed in the client (and constructed them this way in tests).